### PR TITLE
i18n - compile time inlining source map support

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: tools/gulp-tasks/format.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ const srcsToFmt = [
   '!**/dist/**',
   '!**/built/**',
   '!shims_for_IE.js',
+  // Ignore $localize test files
+  '!packages/localize/**/test_files/**',
```
Comment (gkalpak) [inline tools/gulp-tasks/format.js]: Why is this needed?

Comment (petebacondarwin) [reply tools/gulp-tasks/format.js]: Because I changed the integration test to run against a real JS file that is generated by compiling some TS. And the clang formatter doesn't like it.

File: packages/localize/src/tools/test/translate/integration/source_map_utils.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        this.originalSources.set(m.source, originalSource);
+      }
+    });
+  }
+
+  getMappedSegments(): [string, string][] {
+    return this.mappings.map((_, i) => [this.getOriginalText(i), this.getGeneratedText(i)]);
+  }
+
+  // dumpMappings(): void {
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/test/translate/integration/source_map_utils.ts]: Is this function still needed?

**Review by AndrewKushnir — APPROVED** [review]
LGTM (just one nit) 👍

File: packages/localize/src/tools/test/translate/integration/source_map_utils.ts (hunk 2 above)
Comment (petebacondarwin) [reply packages/localize/src/tools/test/translate/integration/source_map_utils.ts]: Hi @AndrewKushnir - thanks for the review.
This function is super useful for debugging the tests - and it is complex enough that I don't want to write it again every time I need to use it. But equally it is not being used right now o I didn't want to leave it in the code. (Apart from anything, it causes lint checks to fail).
What do you think is the best approach in this case?

**Comment by petebacondarwin** [comment]
This approach needs to be updated to use something similar to https://github.com/angular/angular/pull/35132

**Comment by petebacondarwin** [comment]
This will not be the approach taken when if we do sort out source-maps...
